source/common/rate_limiter_impl.cc

```diff
@@ -78,6 +78,43 @@ void LinearRateLimiter::releaseOne() {
   acquired_count_--;
 }
 
+LinearRampingRateLimiterImpl::LinearRampingRateLimiterImpl(Envoy::TimeSource& time_source,
+                                                           const std::chrono::nanoseconds ramp_time,
+                                                           const Frequency frequency)
+    : RateLimiterBaseImpl(time_source), ramp_time_(ramp_time), frequency_(frequency) {
+  if (frequency_.value() <= 0) {
+    throw NighthawkException("frequency must be > 0");
+  }
+  if (ramp_time <= 0ns) {
+    throw NighthawkException("duration must be positive");
+  }
+}
+
+bool LinearRampingRateLimiterImpl::tryAcquireOne() {
+  if (acquireable_count_) {
+    acquired_count_++;
+    return acquireable_count_--;
+  }
+  const auto elapsed_time = elapsed() + 1ns;
```

Comment:
Ugh, I have to refresh my memory. Backing it out it seems to make the first release timing
in tests go off, e.g. we start at `1001` instead of `1000`, I need to dive in to figure out why that was again. The test used to have a 1 ms resolution, possibly that has something to do with it.
I'll figure this out and get back to this.

Reply:
Please see 5cfa171ba9bcc920fe36ad220b81d0e092ecd24b
Summary:

- Backed out the 1 ns adjustment
- Add comment with explanation & amend test expectations
- Also, fix a TODO and make the test use microsecond step resolution now that we can (as of https://github.com/envoyproxy/envoy/pull/9229)